llvm/include/llvm/MC/MCSectionGOFF.h

```diff
@@ -50,26 +53,26 @@ class MCSectionGOFF final : public MCSection {
   friend class MCContext;
   friend class MCSymbolGOFF;
 
-  MCSectionGOFF(StringRef Name, SectionKind K, GOFF::SDAttr SDAttributes,
-                MCSectionGOFF *Parent)
-      : MCSection(SV_GOFF, Name, K.isText(), /*IsVirtual=*/false, nullptr),
+  MCSectionGOFF(StringRef Name, SectionKind K, bool IsVirtual,
```

Comment:
If it's BSS, call it BSS.

Virtual seems a weird naming convention when the initial Mach-O MC code was added. New code doesn't need to endorse the weird naming.

Reply:
It's not actually about BSS (those are named BSS - but they're not virtual).  The "virtual" marker is for sections that represent those GOFF symbols that do not themselves contain text, but have nested GOFF sections which do.  For example the section representing a SD GOFF section does not contain text directly, but it does contain ED sections; and sections representing a ED symbol with the merge attribute do not contain text directly, but contain PR sections.

Reply:
Exactly. The idea is to trigger the check in [MCAssembler::writeSectionData()](https://github.com/llvm/llvm-project/blob/main/llvm/lib/MC/MCAssembler.cpp#L796) for those sections which are not allowed to contain text.